### PR TITLE
[dashboard] fix pending state in Teams

### PR DESCRIPTION
**Pull request by AlexTugarev**
The pending purchase state ...

* should expire after a short while
* should not be persisted until the payment endpoint is hit
* should be updated (removed) on forth/back navigation

### CODE REVIEW

File: components/dashboard/src/settings/Teams.tsx [hunk 1]
```diff
@@ -49,29 +50,73 @@ function AllTeams() {
     const [inviteMembersModal, setInviteMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
     const [addMembersModal, setAddMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
 
-    const [pendingPlanPurchase, setPendingPlanPurchase] = useState<{ plan: Plan } | undefined>(getLocalStorageObject<{ plan: Plan }>("pendingPlanPurchase"));
-    const [pendingSlotsPurchase, setPendingSlotsPurchase] = useState<{ ts: TeamSubscription } | undefined>(getLocalStorageObject<{ ts: TeamSubscription }>("pendingSlotsPurchase"));
+    const restorePendingPlanPurchase = () => {
+        console.log("restorePendingPlanPurchase")
```
Comment (svenefftinge) [inline components/dashboard/src/settings/Teams.tsx:54]: leftover

File: components/dashboard/src/settings/Teams.tsx [hunk 2]
```diff
@@ -49,29 +50,73 @@ function AllTeams() {
...
+    const pendingPlanPurchasePoller = useRef<Disposable>();
+    const pendingSlotsPurchasePoller = useRef<Disposable>();
+
+    const cancelPollers = () => {
+        pendingPlanPurchasePoller.current?.dispose();
+        pendingSlotsPurchasePoller.current?.dispose();
+    }
 
     useEffect(() => {
+        console.log("useEffect pendingPlanPurchase")
```
Comment (svenefftinge) [inline components/dashboard/src/settings/Teams.tsx:92]: leftover

File: components/dashboard/src/settings/Teams.tsx (hunk 1 above)
Comment (svenefftinge) [reply components/dashboard/src/settings/Teams.tsx:54]: it runs 6 times after reload

**Review by svenefftinge — COMMENTED** [review]
The poller management looks very complicated and from looking at the code I cannot understand the logic behind it.
It tested it on https://at-teams.staging.gitpod-io-dev.com/teams and the pending state never got resolved. Not sure what's going on there, though.

File: components/dashboard/src/settings/Teams.tsx (hunk 1 above)
Comment (AlexTugarev) [reply components/dashboard/src/settings/Teams.tsx:54]: ah, indeed. will add a check for `pendingSlotsPurchase`. 

though, it's uncritical. `pendingSlotsPurchase` is updated a single time on reload.

**Comment by AlexTugarev** [comment]
> The poller management looks very complicated and from looking at the code I cannot understand the logic behind it.

@svenefftinge, the polling machinery was copied from old dashboard. some parts are removed, e.g. warning calls. 

the new thing is about the persisted state of pending purchases, which indeed looks complicated :-( 

> the pending state never got resolved 

did you receive the confirmation email? if not, that's just the testing account of chargebee being super slow.

**Comment by svenefftinge** [comment]
> did you receive the confirmation email? if not, that's just the testing account of chargebee being super slow.

Yes, I received it now. Then I went back to the screen and it was still flshing in pending state. After reload it went green.

**Review by svenefftinge — APPROVED** [review]

**Pull request merged**